crashtracker/src/crash_info.rs

```diff
@@ -105,6 +107,11 @@ impl CrashInfo {
         Ok(())
     }
 
+    pub fn set_metadata(&mut self, metadata: CrashtrackerMetadata) -> anyhow::Result<()> {
+        anyhow::ensure!(self.metadata.is_none());
+        self.metadata = Some(metadata);
+        Ok(())
+    }
     pub fn set_siginfo(&mut self, siginfo: SigInfo) -> anyhow::Result<()> {
```

Comment:
Minor: Kinda surprised rustfmt doesn't add a newline here?

```suggestion
    pub fn set_metadata(&mut self, metadata: CrashtrackerMetadata) -> anyhow::Result<()> {
        anyhow::ensure!(self.metadata.is_none());
        self.metadata = Some(metadata);
        Ok(())
    }
    
    pub fn set_siginfo(&mut self, siginfo: SigInfo) -> anyhow::Result<()> {
```